make_mesh.py

```diff
@@ -0,0 +1,256 @@
+	
+	Returns:
+	---------
+	fin : (X,Y,Z) array
+		manipulated data matrix to be used for marching cube
+	iso_surface : float
+		corresponding iso surface value to use for marching cube
+	
+	"""
+	x,y,z = numpy.shape(img_data)
+	x = floor(0.5* x)
+	y = floor(0.5*y)
+	z = floor(0.5*z)
+
+	#Keep original array
+	origin = numpy.copy(img_data)
+	
+	#Fill in the holes within the boundary of the eroded mask
+	img_data[(img_data > 0) & (mask == 1)] = treshhold
+	
+	#To create a smooth inner data matrix that has the overall mean value as max value, calculate value needed to multiply with mask
+	substitute_value = float(treshhold) / float(numpy.max(mask))
+
+	#Replace all inner values of the original data matrix with the smoothed mask multiplied by substitute
+	img_data[numpy.nonzero(mask)]=numpy.multiply(mask[numpy.nonzero(mask)],substitute_value)
+
+	#Choose the isosurface value slightly below the substitute value. This will ensure a singular mesh and a smooth surface in case mask is visible.
+	iso_surface = float(treshhold) / float(1.05)
+	
+	#The final data matrix consists of the maximum values in either the smoothed mask or the original. This ensures that either the original data will be taken or, in case
+	#where the original data matrix will have too low intensities for marching cube to detect (i.e creating wholes in the mesh), the smoothed mask will be taken,resulting in smooth surface
+	fin = numpy.copy(img_data)
+	fin[numpy.nonzero(img_data)] = numpy.fmax(img_data[numpy.nonzero(img_data)],origin[numpy.nonzero(img_data)])
+	return(fin,iso_surface);
+
+#Either take boundary from supplied mask or if not specified, from image directly
+def cut_img_mas(file_input,file_output,size,axis,direction,mask = None):
+	"""
+	Function to trim data matrix before mesh creation. Reads in nifti file and saves the trimmed image as nifti file.
+
+	Parameters:
+	-----------
+	file_input: str
+		file name of image to be loaded and cut (nifti format)
+	file_output: str
+		output file name
+	size : int
+		number of voxels to trim
+	axis : int
+		axis along which to trim (0,1,2)
+	direction : int
+		either trim form beginning af axis (0) inwards or from end of axis (1) inwards
+	mask : (X,Y,Z) array
+		optional. If given, boundary of image will be determined from the mask. Needed if image has boundary with non-zero entries
+
+	"""
+
+	path = os.path.abspath('.')
+	path = path + '/'
+	img= nibabel.load(path+file_input)
+	img_data = img.get_fdata()
+	header=img.header.copy()
+	if (mask != None):
+		mask= nibabel.load(mask)
+		mask_data = mask.get_fdata()
+		box = get_bounding_slices(mask_data)
+	else:
+		box = get_bounding_slices(img_data)
+	img_data = cut_img(img_data,box,size,axis,direction)
+	img_nifti=nibabel.Nifti1Image(img_data,None,header=header)
+	nibabel.save(img_nifti,file_output)
+	return
+
+#Define the boundin:g box of the data matrix. 
+def get_bounding_slices(img):
+	"""
+	Function to determine the boundaries of the given image.
```

Comment:
It is obvious from the syntax that this is a function. "Determine the boundaries of the given image." is both shorter and a real sentence :) (this happens in multiple places).